unit-tests/run-unit-tests.py

```diff
@@ -293,15 +293,22 @@ def __init__( self, source, line_prefix ):
 
         # Parse the python
         regex = r'^' + line_prefix + r'(\S+)((?:\s+\S+)*?)\s*(?:#\s*(.*))?$'
+        each_syntax = re.compile( r'^each\(.+\)$', re.IGNORECASE )
         for context in file.grep( regex, source ):
             match = context['match']
             directive = match.group(1)
+            text_params = match.group(2)
             params = [s for s in context['match'].group(2).split()]
             comment = match.group(3)
             if directive == 'device':
                 #log.d( '    configuration:', params )
                 if not params:
                     log.e( source + '+' + str(context['index']) + ': device directive with no devices listed' )
+                elif 'each' in text_params.lower():
+                    if len(params) > 1:
+                        log.e( source + '+' + str(context['index']) + ': each() cannot be used in combination with other specs', params )
+                    if not each_syntax.match( text_params ):

```

Comment:
elif

Reply:
I'd prefer if the pattern was inline here: `if not re.fullmatch( r'each\(.+\)', text_params, re.IGNORECASE ):`

Reply:
🤦‍♂️